src/microbe_stage/FluidCurrentDisplay.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Godot;
+
+/// <summary>
+///   Displays fluid currents in the microbe stage
+/// </summary>
+public partial class FluidCurrentDisplay : GpuParticles3D
+{
+#pragma warning disable CA2213
+    private ShaderMaterial material = null!;
+
+    private MicrobeStage stage = null!;
+#pragma warning restore CA2213
+
+    private StringName gameTimeParameterName = new("gameTime");
+    private StringName speedParameterName = new("speed");
+    private StringName chaoticnessParameterName = new("chaoticness");
+    private StringName scaleParameterName = new("scale");
+
+    private float time;
+
+    public void Init(MicrobeStage microbeStage)
+    {
+        stage = microbeStage;
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        material = (ShaderMaterial)ProcessMaterial;
+    }
+
+    public override void _Process(double delta)
+    {
+        base._Process(delta);
+
+        GlobalPosition = new Vector3(GlobalPosition.X, 1.0f, GlobalPosition.Z);
+
+        if (!PauseManager.Instance.Paused)
+        {
+            material.SetShaderParameter(gameTimeParameterName, time);
```

Comment:
Is it intentional that this passes in the *old* time? I think I'll swap this around the way I think this should be.